.github/workflows/daily-credits.yaml

```
@@ -0,0 +1,34 @@
+
+name: daily credits build
+
+on:
+  schedule:
+    - cron: "5 0 * * *"
+
+jobs:
+  build_credits:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout latest code
+        uses: actions/checkout@v2.3.1
+        with:
+          fetch-depth: 0
+      - name: Get current date
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+      - name: cache credits
+        id: cache-credits
+        uses: actions/cache@v3.0.0
+        with:
+          key: credits-${{ steps.date.outputs.date }}
+          path: credits.txt
+      - name: update credits
+        uses: actions/github-script@v3
+        with:
+          script: |
+            const path = require('path');
+            const script = require(path.resolve('.github/workflows/update-credits.js'));
+            await script(github, context)
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  
```

We can add this steps to `daily credits build` workflow to push a new build as well

Though it maybe bad idea to push a new build everyday even if number of contributors or their usernames didn't change. So we might want to have some sort of hash check of `credits.txt` for a previous day - and if file didn't change we won't run next steps

```suggestion
       - name: export with Godot Engine
        uses: firebelley/godot-export@v3.0.0
        with:
          create_release: false
          archive_export_output: true
          godot_executable_download_url: https://downloads.tuxfamily.org/godotengine/3.4.4/Godot_v3.4.4-stable_linux_headless.64.zip
          godot_export_templates_download_url: https://downloads.tuxfamily.org/godotengine/3.4.4/Godot_v3.4.4-stable_export_templates.tpz
          relative_export_path: ./
          relative_project_path: ./
      - name: deploy to itch.io
        uses: josephbmanley/butler-publish-itchio-action@master
        env:
          BUTLER_CREDENTIALS: ${{ secrets.ITCH_API_KEY }}
          CHANNEL: html
          ITCH_GAME: a-little-game-called-mario
          ITCH_USER: iznaut
          PACKAGE: HTML5.zip
          VERSION: credits-update
```

---

We can have few outputs in `date` step 
```suggestion
  - name: Get current date
    id: date
    run: |
       echo ::set-output name=date::"$(date +'%Y-%m-%d')"
       echo ::set-output name=yesterday::"$(date --date='yesterday' +'%Y-%m-%d')"
```

And have additional step to get credits for previous date cache
```suggestion
      - name: yesterday credits
        id: yesterday-credits
        uses: actions/cache@v3.0.0
        with:
          key: credits-${{ steps.date.outputs.yesterday }}
          path: credits-yesterday.txt
```

Then we can check for their hash sums in our build steps
```suggestion
if: ${{ hashFiles('credits.txt') != hashFiles('credits-yesterday.txt') }}
```

But this might fail if `credits-yesterday.txt` is not exist, so we want another additional step
```suggestion
-  name: stub yesterday credits
   id: stub-yesterday-credits
   if: ${{ steps.yesterday-credits.outputs.cache-hit != 'true' }}
   run: touch credits-yesterday.txt
```

---

It does not seem possible to restore a file from a cache item using a different path
```suggestion
        - name: yesterday credits
        id: yesterday-credits
        uses: actions/cache@v3.0.0
        with:
          key: credits-${{ steps.date.outputs.yesterday }}
          path: credits-yesterday.txt
```
This action will search for a file called `credits-yesterday.txt` in the corresponding cache item.

Sadly i cannot think of an alternative way to solve this problem.

---

@HanLap
Oh so you want to say, that both `key` and `path` used to determine what file was cached?
Yeah, I guess I I misunderstood that and I was thinking that `key` - is determine the file and a `path` just used to save/restore file for that `key`

So in that case, we can restore `yesterday-credits.txt` first, then rename it, and only then restore `credits.txt`:
```
- id: date
   run: |
      echo ::set-output name=date::"$(date +'%Y-%m-%d')"
      echo ::set-output name=yesterday::"$(date --date='yesterday' +'%Y-%m-%d')"

- id: restore-yesterday-credits
  uses: actions/cache@v3.0.0
  with:
    key: credits-${{ steps.date.outputs.yesterday }}
    path: credits.txt
      
- id: stub-yesterday-credits
   if: ${{ steps.restore-yesterday-credits.outputs.cache-hit != 'true' }}
   run: touch credits.txt
     
- run: mv credits.txt credits-yesterday.txt

- id: cache-credits
  uses: actions/cache@v3.0.0
  with:
    key: credits-${{ steps.date.outputs.date }}
    path: credits.txt
    
- name: update credits
   uses: actions/github-script@v3
   with:
     script: |
       const path = require('path');
       const script = require(path.resolve('.github/workflows/update-credits.js'));
       await script(github, context)
       
- if: ${{ hashFiles('credits.txt') != hashFiles('credits-yesterday.txt') }}
  #BUILD
```